testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketShutdownOutputBySelfTest.java

```diff
@@ -140,5 +219,21 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
         public void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Exception {
             queue.offer(msg.readByte());
         }
+
+        private void drainWritabilityQueue() throws InterruptedException {
+            while ((writabilityQueue.poll(100, TimeUnit.MILLISECONDS)) != null) {
+            }
+        }
+
+        void assertWritability(boolean isWritable) throws InterruptedException {
+            try {
+                Boolean writability = writabilityQueue.takeLast();
+                assertEquals(isWritable, writability);
+                // TODO(scott): why do we get multiple writability changes here ... race condition?
```

Comment:
@normanmaurer - note that while writing outside the event loop I'm getting more writability change notifications than expected. If you would also only expect to get two notifications (not writable after the write, then writable after the flush) then I can open an issue and investigate as a followup.